src/main/commands/scan.yml

```diff
@@ -0,0 +1,42 @@
+description: Detect bugs and vulnerabilities
+parameters:
+  sonar_token_variable_name:
+    description: the name of the environment variable where the SonarCloud API token is stored 
+    default: SONAR_TOKEN
+    type: env_var_name
+  cache_version:
+    description: increment this value if the cache is corrupted and you want to start with a clean cache
+    default: 1
+    type: integer
+steps:
+  - run:
+      name: Create cache directory if it doesn't exist
+      command: mkdir -p /tmp/cache/scanner
+  - restore_cache:
+      keys:
+        - v<<parameters.cache_version>>-sonarcloud-scanner-4.1.0.1829
```

Comment:
The scanner version does not need to be part of the cache key. The cached files are compatible with all versions of scanners.

Reply:
That looks like a cache for the (versioned) sonar-scanner executable, not necessarily a cache of generated files

Reply:
Indeed, the cache will hold the scanner executable, and the plugins downloaded by the scanner. It is not a big deal if we remove it, but if we ever want to use another scanner executable version then we should change the cache name.

Reply:
@tom-vanbraband-sonarsource, have you been able to get it to avoid downloading plugins by caching?  It seems that even when I'm using a cache, sonar-scanner still downloads plugins every time, and it's the longest 17 seconds of my life.

Reply:
no I failed to make it work.

Reply:
Hi @jrnail23, the SONAR_USER_HOME variable determines among other things the directory where your downloaded plugins are stored. So you can use that to make sure that your plugins are stored within the circleci cache directory.